api/docs/reference/search_algorithm.md

```diff
@@ -0,0 +1,226 @@
+  catalogued in Openverse.
+- "Field": The individual queryable elements of a document. In the context of
+  Openverse, these may be textual, numerical, or keywords.
+- "Result relevance": An individual result is "relevant" to a search when it
+  matches the expectations of the user for a given query. A result is irrelevant
+  if it does not match the expectations of the user for that same query. For
+  example, the query "bird watch" may produce pictures of a wrist watch with a
+  bird clockface illustration _or_ could surface pictures related to the
+  activity also known as "birding" (bird watching), due to stemming. In the case
+  of this specific query, "bird watching" may not be relevant, despite being a
+  technically correct match for the query given Openverse's current index
+  configuration. Other relevancy issues may be caused by descriptions that are
+  not related to the contents of an image. This often happens on Flickr where
+  users sometimes include blog-like text in the description of an image that
+  references things that happened outside of the context of the image itself.
+- "Result quality": A combination of relevance and other factors like the actual
+  perceived "quality" of a given work. A work may be directly relevant to a
+  particular query but be of low quality. Quality is subjective, though there
+  may be certain characteristics that are broadly applicable to some subset of
+  searches.
+
+## Technology
+
+Openverse uses Elasticsearch's
+[full text indexing and search capabilities](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/full-text-queries.html).
+We currently rely heavily on Elasticsearch's default behaviours in many aspects
+of our search including Elasticsearch's default stemming configuration, aside
+from small adjustments documented in the
+[text analysis and stemming](#text-analysis/tokenization) section below. The
+"raw" index configuration can be found in the `es_mapping.py` module (see link
+in [resources](#resources)). Information for how to understand the configuration
+can be found in the
+[Elasticsearch documentation for index configuration](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/index-modules.html).
+
+> **Note**: We also apply cluster-level configurations as part of our
+> Elasticsearch deployment. These are intentionally not covered here as they
+> primarily deal with cluster performance and are irrelevant to the way searches
+> are executed.
+
+## Text analysis/tokenization
+
+> **Note**: A general understanding of
+> [full-text search (Wikipedia)](https://en.wikipedia.org/wiki/Full-text_search)
+> and the concepts of
+> ["stemming" (Wikipedia)](https://en.wikipedia.org/wiki/Stemming) and
+> ["tokenization" (Wikipedia)](https://en.wikipedia.org/wiki/Lexical_analysis#Tokenization)
+> will be useful for understanding this section.
+
+Text analysis or tokenization is the broad process Elasticsearch follows to
```

Comment:
I think the phrase "derive the index tokens" can be difficult for someone very new to search in general. If that is the target audience for this doc, we could add something like the following: 

To search texts, Elasticsearch builds an index structure where keys are tokens (words) and values are the items that contain these tokens: (_Small diagrams illustrating the index below have helped me greatly to understand what an index is_):
{
"cat": [image1, image4, image5],
"bird": [image2, image4]
}
To create such an index in Openverse, we split the original texts (in Openverse case, it can be the media title, description, tags) into tokens.

Reply:
I'd like to avoid trying to document the specific ways in which full text search work, which is why I linked the handful of Wikipedia pages above. If I added "Index" as a term in the terms list with a link to the [inverted index](https://en.wikipedia.org/wiki/Inverted_index) Wikipedia entry, would that be sufficient? Something like:

> Index: A data structure used by Elasticsearch to quickly identify the queryable terms for a document. Specifically, an [inverted index](https://en.wikipedia.org/wiki/Inverted_index).

As for who the intended audience is, I wrote this assuming a relatively technical audience, mostly because it is documentation in the codebase rather than documentation we would present on the website. I tried to make it accessible for technical people without prior knowledge of how full-text search works by adding links to helpful resources, but making it accessible to a fully non-technical audience would be a tall order.